src/AuthorizationService.php

```diff
@@ -0,0 +1,73 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         1.0.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Authorization;
+
+use Authorization\Policy\BeforePolicyInterface;
+use Authorization\Policy\Exception\MissingMethodException;
+use Authorization\Policy\ResolverInterface;
+
+class AuthorizationService implements AuthorizationServiceInterface
+{
+    /**
+     * Authorization policy resolver.
+     *
+     * @var \Authorization\Policy\ResolverInterface
+     */
+    protected $resolver;
+
+    /**
+     *
+     * @param \Authorization\Policy\ResolverInterface $resolver Authorization policy resolver.
+     */
+    public function __construct(ResolverInterface $resolver)
+    {
+        $this->resolver = $resolver;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function can(IdentityInterface $user, $action, $resource)
+    {
+        $policy = $this->resolver->getPolicy($resource);
+
+        if ($policy instanceof BeforePolicyInterface && !$policy->before($user, $resource)) {
```

Comment:
Should we allow a policy's before() method to return true to allow all operations?

Reply:
Should the before hook know the action being take as well?

Reply:
Maybe a before method could allow the authorization check if a policy method hasn't been defined.
If a method exists the result of that would be returned.